src/WebJobs.Script/SanitizerRegex.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.WebJobs.Logging
+{
+    internal partial class SanitizerRegex
+    {
+        // Pattern of format : "<protocol>://<username>:<password>@<address>:<port>"
+        private static readonly string Pattern = @"
+                                                \b([a-zA-Z]+)            # Capture protocol
+                                                :\/\/                    # '://'
+                                                ([^:/\s]+)               # Capture username
+                                                :                        # ':'
+                                                ([^@/\s]+)               # Capture password
+                                                @                        # '@'
+                                                ([^:/\s]+)               # Capture address
+                                                :                        # ':'
+                                                ([0-9]+)\b               # Capture port number
+                                            ";
+
+#if NET8_0_OR_GREATER
+        [GeneratedRegex(Pattern, RegexOptions.IgnoreCase | RegexOptions.IgnorePatternWhitespace, "en-US")]
+        internal static partial Regex Regex();
```

Comment:
Using .Net feature `GeneratedRegex` for performance - https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-source-generators?pivots=dotnet-8-0#source-generation